src/test/java/com/streisky/virtualshowcasebackend/controller/product/ProductControllerTest.java

```diff
@@ -0,0 +1,159 @@
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.json.JacksonTester;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import static com.streisky.virtualshowcasebackend.constants.ProductTestConstants.PAGE_REQUEST_DEFAULT;
+import static com.streisky.virtualshowcasebackend.constants.ProductTestConstants.PRODUCT_DTO_ACTIVE;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@AutoConfigureJsonTesters
+@ActiveProfiles("test")
+class ProductControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private JacksonTester<ProductDTO> productDTORequestJson;
+
+    @Autowired
+    private JacksonTester<ProductDTO> productDTOResponseJson;
+
+    @Autowired
+    private JacksonTester<Page<ProductDTO>> productDTOListResponseJson;
+
+    @MockBean
+    private ProductService productService;
+
+    public static final String PRODUCT_URL = "/products";
+    public static final String PRODUCT_URL_WITH_ID = PRODUCT_URL + "/{id}";
+
+    @Test
+    @WithMockUser
+    void requestPostWithInvalidBodyTest() throws Exception {
+        MockHttpServletResponse response = mockMvc.perform(MockMvcRequestBuilders.post(PRODUCT_URL)).andReturn().getResponse();
+        Assertions.assertEquals(HttpStatus.BAD_REQUEST.value(), response.getStatus());
+    }
+
+    @Test
+    @WithMockUser
+    void requestPostWithValidBodyTest() throws Exception {
+        Mockito.when(productService.save(Mockito.any())).thenReturn(PRODUCT_DTO_ACTIVE);
+
+        MockHttpServletResponse response = mockMvc
+                .perform(
+                        MockMvcRequestBuilders
+                                .post(PRODUCT_URL)
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(productDTORequestJson.write(PRODUCT_DTO_ACTIVE).getJson())
+                )
+                .andReturn().getResponse();
+        String jsonExpected = productDTOResponseJson.write(PRODUCT_DTO_ACTIVE).getJson();
+
+        Assertions.assertEquals(HttpStatus.CREATED.value(), response.getStatus());
+        Assertions.assertEquals(JsonParser.parseString(jsonExpected), JsonParser.parseString(response.getContentAsString()));
+    }
+
+    @Test
+    @WithMockUser
+    void requestPutWithInvalidBodyTest() throws Exception {
```

Comment:
I think names could be a little more descriptive. For example, what part of the body is invalid, and what is the expected result?

There are some patterns for naming that use underscore (_) to better separate contexts.
I particularly like using `Should_ExpectedBehavior_When_StateUnderTest`, but that is up to discussion.

Reply:
Renamed all tests methods.